### PR TITLE
[Serve][Doc] Add loadbalancer and monitoring section for K8s

### PR DESCRIPTION
**Pull request by sihanwang41**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/production-guide/kubernetes.md [hunk 1]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
 In most use cases, it is recommended to enable Kubernetes autoscaling to fully utilize the resources in your cluster. If you are using GKE, you can utilize the AutoPilot Kubernetes cluster. For instructions, see [Create an Autopilot Cluster]((https://cloud.google.com/kubernetes-engine/docs/how-to/creating-an-autopilot-cluster)). For EKS, you can enable Kubernetes cluster autoscaling by utilizing the Cluster Autoscaler. For detailed information, see [Cluster Autoscaler on AWS](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md). To understand the relationship between Kubernetes autoscaling and Ray autoscaling, see [Ray Autoscaler with Kubernetes Cluster Autoscaler](kuberay-autoscaler-with-ray-autoscaler).
 :::
 
+## Load balancer
+You can setup ingress to expose your Serve application to the internet. See [example](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice-alb-ingress.yaml)
```
Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:259]:
```suggestion
You can set up an ingress to expose your Serve application to the internet. See [this configuration](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice-alb-ingress.yaml) for an example.
```

File: doc/source/serve/production-guide/kubernetes.md [hunk 2]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+
+:::{note}
+- Ray Serve provisions an HTTP proxy on every node, allowing you to use `/-/routes` as the endpoint for node health checks.
```
Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:262]:
```suggestion
- Ray Serve runs an HTTP proxy on every node. Each proxy exposes contains a `/healthz` endpoint that you can query to check its health.
```

File: doc/source/serve/production-guide/kubernetes.md [hunk 3]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+- Ray Serve uses port 8000 as the default HTTP proxy traffic port. You can change the port by setting `http_options` in the Serve config. Learn more details [here](serve-multi-application)
```
Comment (angelinalg) [inline doc/source/serve/production-guide/kubernetes.md:263]:
```suggestion
- Ray Serve uses port 8000 as the default HTTP proxy traffic port. You can change the port by setting `http_options` in the Serve config. See [Deploy Multiple Applications](serve-multi-application) for more details.
```

File: doc/source/serve/production-guide/kubernetes.md [hunk 4]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+:::
+
+## Monitoring
+You can monitor your Serve application using the Ray dashboard.
```
Comment (angelinalg) [inline doc/source/serve/production-guide/kubernetes.md:267]:
```suggestion
Monitor your Serve application using the Ray Dashboard.
```

File: doc/source/serve/production-guide/kubernetes.md (hunk 3 above)
Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:263]:
```suggestion
- Ray Serve uses port 8000 as the default HTTP proxy traffic port. You can change the port by setting `http_options` in the Serve config. Learn more details [here](serve-multi-application).
```

File: doc/source/serve/production-guide/kubernetes.md [hunk 5]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+Learn more about how to configure & manage dashboard [here](observability-configure-manage-dashboard)
```
Comment (angelinalg) [inline doc/source/serve/production-guide/kubernetes.md:268]:
```suggestion
Learn more about how to configure and manage Dashboard [here](observability-configure-manage-dashboard)
```

Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:268]:
```suggestion
Learn more about how to configure and manage the dashboard [here](observability-configure-manage-dashboard).
```

File: doc/source/serve/production-guide/kubernetes.md [hunk 6]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+Learn ray serve dashboard [here](serve-monitoring)
```
Comment (angelinalg) [inline doc/source/serve/production-guide/kubernetes.md:269]:
```suggestion
Learn about the Ray Serve Dashboard [here](serve-monitoring)
```

Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:269]:
```suggestion
Learn more about the dashboard's Ray Serve section [here](serve-monitoring).
```

File: doc/source/serve/production-guide/kubernetes.md [hunk 7]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+Learn how to setup Grafana dashboard [here](https://ray-project.github.io/kuberay/guidance/prometheus-grafana/)
```
Comment (shrekris-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:270]:
```suggestion
Learn how to set up the Grafana dashboard [here](https://ray-project.github.io/kuberay/guidance/prometheus-grafana/).
```

Comment (angelinalg) [inline doc/source/serve/production-guide/kubernetes.md:270]:
```suggestion
Learn how to set up Grafana for Dashboard [here](https://ray-project.github.io/kuberay/guidance/prometheus-grafana/)
```

File: doc/source/serve/production-guide/kubernetes.md (hunk 1 above)
Comment (angelinalg) [inline doc/source/serve/production-guide/kubernetes.md:259]:
```suggestion
Set up ingress to expose your Serve application to the internet. ([Example])(https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice-alb-ingress.yaml)
```

Comment (angelinalg) [reply doc/source/serve/production-guide/kubernetes.md:259]: Hope that syntax works.

**Review by angelinalg — APPROVED** [review]

File: doc/source/serve/production-guide/kubernetes.md [hunk 8]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
 In most use cases, it is recommended to enable Kubernetes autoscaling to fully utilize the resources in your cluster. If you are using GKE, you can utilize the AutoPilot Kubernetes cluster. For instructions, see [Create an Autopilot Cluster]((https://cloud.google.com/kubernetes-engine/docs/how-to/creating-an-autopilot-cluster)). For EKS, you can enable Kubernetes cluster autoscaling by utilizing the Cluster Autoscaler. For detailed information, see [Cluster Autoscaler on AWS](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md). To understand the relationship between Kubernetes autoscaling and Ray autoscaling, see [Ray Autoscaler with Kubernetes Cluster Autoscaler](kuberay-autoscaler-with-ray-autoscaler).
 :::
 
+## Load balancer
+Set up ingress to expose your Serve application to the internet. See [this configuration](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice-alb-ingress.yaml)
```
Comment (akshay-anyscale) [inline doc/source/serve/production-guide/kubernetes.md:259]: What if you want an internal service (not exposed to the internet)?

**Review by akshay-anyscale — COMMENTED** [review]
the ray_v1alpha1_rayservice.yaml needs to be updated to use serveConfigV2

**Comment by sihanwang41** [comment]
> ray_v1alpha1_rayservice.yaml

Hi @akshay-anyscale, the corresponding link can be updated after the new kuberay version is released. Currently  ray_v1alpha1_rayservice.yaml in latest master is already using service v2 config. Normally we will have doc version audit in the ray doc everytime we have a new version of kuberay release.

File: doc/source/serve/production-guide/kubernetes.md (hunk 8 above)
Comment (sihanwang41) [reply doc/source/serve/production-guide/kubernetes.md:259]: Rephrased. Basically it depends on network config whether your ingress can be connected by external.

**Comment by akshay-anyscale** [comment]
> > ray_v1alpha1_rayservice.yaml
> 
> Hi @akshay-anyscale, the corresponding link can be updated after the new kuberay version is released. Currently ray_v1alpha1_rayservice.yaml in latest master is already using service v2 config. Normally we will have doc version audit in the ray doc everytime we have a new version of kuberay release.

Got it, don't those changes need to be done before Ray 2.6 release? CC: @kevin85421

File: doc/source/serve/production-guide/kubernetes.md [hunk 9]
```diff
@@ -255,6 +255,21 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
 
+## Load balancer
+Set up ingress to expose your Serve application with a load balancer. See [this configuration](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice-alb-ingress.yaml)
+
+:::{note}
+- Ray Serve runs HTTP proxy on every node, allowing you to use `/-/routes` as the endpoint for node health checks.
+- Ray Serve uses port 8000 as the default HTTP proxy traffic port. You can change the port by setting `http_options` in the Serve config. Learn more details [here](serve-multi-application).
+:::
+
+## Monitoring
```
[the remark this answers is not in the document]
Comment (akshay-anyscale) [reply doc/source/serve/production-guide/kubernetes.md:266]: Let's also add how to get to the kuberay logs for troubleshooting

[the remark this answers is not in the document]
Comment (sihanwang41) [reply doc/source/serve/production-guide/kubernetes.md:266]: added logging with Loki link and trouble shooting notes.

[the remark this answers is not in the document]
Comment (akshay-anyscale) [reply doc/source/serve/production-guide/kubernetes.md:266]: thanks for updating! Do we have anything we recommend for metrics too? Btw I noticed this guide as well - https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/logging.html . @scottsun94 which one do you usually recommend?

[the remark this answers is not in the document]
Comment (scottsun94) [reply doc/source/serve/production-guide/kubernetes.md:266]: For log aggregation tools, we list a few [here](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/logging.html): [Fluent Bit](https://docs.fluentbit.io/manual), [Vector](https://vector.dev/), [Fluentd](https://docs.fluentd.org/), [Filebeat](https://www.elastic.co/guide/en/beats/filebeat/7.17/index.html), and [Promtail](https://grafana.com/docs/loki/latest/clients/promtail/). We use Fluent Bit as an example but any of them should work and it depends on the users' preferences.

Serve doc have instructions for how to use Promtail with Loki which is good. It doesn't expand on how to persist the logs using Promtail. Here is my suggestion for how to reorganize the links in this section:
- people need to learn how to export metrics first
- then about dashboard
- lastly log persistence

```
Learn how to export metrics and set up the Grafana [here](https://ray-project.github.io/kuberay/guidance/prometheus-grafana/)
Learn more about how to configure and manage Dashboard [here](observability-configure-manage-dashboard).
Learn about the Ray Serve Dashboard [here](serve-monitoring).
Learn about the [Ray Serve logs](serve-logging) and how to [persistent logs](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/logging.html )on Kubernetes clusters.
```

**Comment by sihanwang41** [comment]
> > > ray_v1alpha1_rayservice.yaml
> > 
> > 
> > Hi @akshay-anyscale, the corresponding link can be updated after the new kuberay version is released. Currently ray_v1alpha1_rayservice.yaml in latest master is already using service v2 config. Normally we will have doc version audit in the ray doc everytime we have a new version of kuberay release.
> 
> Got it, don't those changes need to be done before Ray 2.6 release? CC: @kevin85421

Yeah, we can update the doc link after the kuberay 0.6.0 version is out. (Don't want to use nightly link in 2.6 release)

File: doc/source/serve/production-guide/kubernetes.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (sihanwang41) [reply doc/source/serve/production-guide/kubernetes.md:266]: Updated!

File: doc/source/serve/production-guide/kubernetes.md [hunk 10]
```diff
@@ -255,6 +255,26 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+Monitor your Serve application using the Ray Dashboard.
+- Learn more about how to configure and manage Dashboard [here](observability-configure-manage-dashboard).
+- Learn about the Ray Serve Dashboard [here](serve-monitoring).
+- Learn how to set up the Grafana for Dashboard [here](https://ray-project.github.io/kuberay/guidance/prometheus-grafana/)
```
Comment (scottsun94) [inline doc/source/serve/production-guide/kubernetes.md:270]: It relies on both Prometheus and Grafana. How about this?
```suggestion
- Learn how to set up [Prometheus](prometheus-setup) and [Grafana](grafana) for Dashboard
```

@sihanwang41 Can we send them to this page instead?  https://docs.ray.io/en/latest/cluster/metrics.html#setting-up-your-prometheus-server

Comment (sihanwang41) [reply doc/source/serve/production-guide/kubernetes.md]: nice, I wasn't aware of we have this content in the ray doc already.

File: doc/source/serve/production-guide/kubernetes.md [hunk 11]
```diff
@@ -255,6 +255,26 @@ To enable autoscaling in a KubeRay Cluster, you need to set `enableInTreeAutosca
...
+- Ray Serve runs HTTP proxy on every node, allowing you to use `/-/routes` as the endpoint for node health checks.
+- Ray Serve uses port 8000 as the default HTTP proxy traffic port. You can change the port by setting `http_options` in the Serve config. Learn more details [here](serve-multi-application).
+:::
+
+## Monitoring
+Monitor your Serve application using the Ray Dashboard.
+- Learn more about how to configure and manage Dashboard [here](observability-configure-manage-dashboard).
+- Learn about the Ray Serve Dashboard [here](serve-monitoring).
+- Learn how to set up [Prometheus](prometheus-setup) and [Grafana](grafana) for Dashboard.
+- Learn about the [Ray Serve logs](serve-logging) and how to [persistent logs](kuberay-logging) on Kubernetes clusters.
```
Comment (scottsun94) [inline doc/source/serve/production-guide/kubernetes.md:271]:
```suggestion
- Learn about the [Ray Serve logs](serve-logging) and how to [persistent logs](kuberay-logging) on Kubernetes.
```

**Review by scottsun94 — APPROVED** [review]

**Review by akshay-anyscale — APPROVED** [review]

**Pull request merged**